File: RxTools-library/src/main/java/com/vondear/rxtools/view/cardstack/tools/RxAdapterStack.java
package com.vondear.rxtools.view.cardstack.tools;

import android.content.Context;
import android.view.LayoutInflater;

import com.vondear.rxtools.view.cardstack.RxCardStackView;

import java.util.ArrayList;
import java.util.List;

/**
 * @author vondear
 */
public abstract class RxAdapterStack<T> extends RxCardStackView.Adapter<RxCardStackView.ViewHolder> {

    private final Context mContext;
    private final LayoutInflater mInflater;
    private List<T> mData;

    public RxAdapterStack(Context context) {
        this.mContext = context;
        this.mInflater = LayoutInflater.from(context);
        this.mData = new ArrayList();
    }

    public void updateData(List<T> data) {
        this.setData(data);
        this.notifyDataSetChanged();
    }

    public void setData(List<T> data) {
        this.mData.clear();
        if (data != null) {
            this.mData.addAll(data);
        }
    }

    public LayoutInflater getLayoutInflater() {
        return this.mInflater;
    }

    public Context getContext() {
        return this.mContext;
    }

    @Override
    public void onBindViewHolder(RxCardStackView.ViewHolder holder, int position) {
        T data = this.getItem(position);
        this.bindView(data, position, holder);
    }

    public abstract void bindView(T data, int position, RxCardStackView.ViewHolder holder);

    @Override
    public int getItemCount() {
        return mData.size();
    }

    public T getItem(int position) {
        return this.mData.get(position);
    }

}

File: RxUI/src/main/java/com/tamsiree/rxui/view/cardstack/tools/RxAdapterStack.kt
package com.tamsiree.rxui.view.cardstack.tools

import android.content.Context
import android.view.LayoutInflater
import com.tamsiree.rxui.view.cardstack.RxCardStackView
import java.util.*

/**
 * @author tamsiree
 * @date 2018/6/11 11:36:40 整合修改
 */
abstract class RxAdapterStack<T>(val context: Context) : RxCardStackView.Adapter<RxCardStackView.ViewHolder>() {
    val layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private val mData: MutableList<T>

    fun updateData(data: List<T>?) {
        setData(data)
        notifyDataSetChanged()
    }

    fun setData(data: List<T>?) {
        mData.clear()
        if (data != null) {
            mData.addAll(data)
//            itemCount=mData.size
        }
    }

    override fun onBindViewHolder(holder: RxCardStackView.ViewHolder, position: Int) {
        val data = getItem(position)
        bindView(data, position, holder)
    }

    abstract fun bindView(data: T, position: Int, holder: RxCardStackView.ViewHolder)

/*    override fun getItemCount(): Int {
        return mData.size
    }*/

    fun getItem(position: Int): T {
        return mData[position]
    }

    init {
        mData = ArrayList()
    }

    override val itemCount: Int
        get() = mData.size


}